### ISSUE TITLE
Inference is slow for me - mask-rcnn tesla k80

### ISSUE THREAD

**Issue by YoadTew** [facebookresearch/detectron2#150]
If you do not know the root cause of the problem / bug, and wish someone to help you, please
include:

I've created a code based on the Detectron2's beginner tutorial in collab. Im running instance segmentation based on the 'COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x/137849600/model_final_f10217.pkl model.
According to the model zoo, inference supposed to take 0.055 s/img, but when I ran my code it takes me about 0.47 s/img.

## To Reproduce

```
# Some basic setup
# Setup detectron2 logger
import detectron2
from detectron2.utils.logger import setup_logger
setup_logger()

# import some common libraries
import numpy as np
import cv2
import glob
import time

# import some common detectron2 utilities
from detectron2.engine import DefaultPredictor
from detectron2.config import get_cfg
from detectron2.utils.visualizer import Visualizer
from detectron2.data import MetadataCatalog

imgs_path = glob.glob('/home/work/data/*.jpg')
cfg = get_cfg()
cfg.merge_from_file("/home/work/detectron2_repo/configs/COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.9  # set threshold for this model
# Find a model from detectron2's model zoo. You can either use the https://dl.fbaipublicfiles.... url, or use the following shorthand
cfg.MODEL.WEIGHTS = "detectron2://COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x/137849600/model_final_f10217.pkl"
predictor = DefaultPredictor(cfg)

for img_path in imgs_path[:20]:
	img_base_name = img_path.split('/')[-1]
	im = cv2.imread(img_path)

	start = time.time()
	outputs = predictor(im)
	end = time.time()

	print('duration:', end-start)

	# We can use `Visualizer` to draw the predictions on the image.
	v = Visualizer(im[:, :, ::-1], MetadataCatalog.get(cfg.DATASETS.TRAIN[0]), scale=1.2)
	v = v.draw_instance_predictions(outputs["instances"].to("cpu"))
	final_img = v.get_image()[:, :, ::-1]

	cv2.imwrite('/home/work/output/detectron2/' + img_base_name, final_img)
```

outputs:
duration: 0.47621917724609375
duration: 0.4896988868713379
duration: 0.4988124370574951
duration: 0.4810659885406494
duration: 0.48575687408447266
duration: 0.49514174461364746
duration: 0.4949469566345215
...

## Expected behavior

I ran the code with tesla-k80, I was expecting the infer time would be close to the baseline showed in the model zoo

## Environment

------------------------  --------------------------------------------------------------------------------------------------
sys.platform              linux
Python                    3.6.8 (default, Jan 14 2019, 11:02:34) [GCC 8.0.1 20180414 (experimental) [trunk revision 259383]]
Numpy                     1.16.4
Detectron2 Compiler       GCC 7.4
Detectron2 CUDA Compiler  10.0
DETECTRON2_ENV_MODULE     <not set>
PyTorch                   1.3.0+cu100
PyTorch Debug Build       False
torchvision               0.4.1+cu100
CUDA available            True
GPU 0,1                   Tesla K80
CUDA_HOME                 /usr/local/cuda
NVCC                      Cuda compilation tools, release 10.0, V10.0.130
Pillow                    6.1.0
cv2                       4.1.0
------------------------  --------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.20.5 (Git Hash 0125f28c61c1f822fd48570b4c1066f96fcb9b2e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CUDA Runtime 10.0
  - NVCC architecture flags: -gencode;arch=compute_35,code=sm_35;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_50,code=compute_50
  - CuDNN 7.6.3
  - Magma 2.5.1
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Wno-stringop-overflow, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=True, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

**Comment by ppwwyyxx** [comment]
The correct way to benchmark inference speed is https://github.com/facebookresearch/detectron2/issues/106#issuecomment-543281947. It will give you a number that's comparable to the one in the model zoo.

**Comment by YoadTew** [comment]
> The correct way to benchmark inference speed is [#106 (comment)](https://github.com/facebookresearch/detectron2/issues/106#issuecomment-543281947). It will give you a number that's comparable to the one in the model zoo.

Thank you for quick response,
I want to use detectron 2 as a library in my project. The mask rcnn segmentation I want to use is a step in my final pipeline.
How can I use the model as a library instead of the command line, and still retain the inference speed comparable to the one in the model zoo?

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/modules/evaluation.html#detectron2.evaluation.inference_on_dataset gives you a more correct calculation of inference speed.
